tools/SetupFlow/DevHome.SetupFlow.ElevatedComponent/IElevatedComponentOperation.cs

```diff
@@ -1,46 +1,51 @@
-/// * The types must be `public sealed class` or `public interface`.
-/// * All public members must use only types that can be projected with CsWinRT.
-/// If the projection fails, it will only produce a warning, but the build is likely to
-/// fail further down the line when attempting to use the types that should have been created.
-/// </remarks>
-public interface IElevatedComponentFactory
-{
-    /// <summary>
-    /// Writes a string to standard output.
-    /// </summary>
-    /// <remarks>
-    /// This is intended for tests only.
-    /// </remarks>
-    public void WriteToStdOut(string value);
-
-    /// <summary>
-    /// Creates an object that can be used to install packages from an elevated context.
-    /// </summary>
-    public ElevatedInstallTask CreateElevatedInstallTask();
-
-    /// <summary>
-    /// Creates the object that will create the Dev Drive on the users system.
-    /// </summary>
-    public DevDriveStorageOperator CreateDevDriveStorageOperator();
-
-    /// <summary>
-    /// Creates an object that can apply a configuration file.
-    /// </summary>
-    public ElevatedConfigurationTask CreateElevatedConfigurationTask();
-}
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using DevHome.SetupFlow.ElevatedComponent.Helpers;
+using Windows.Foundation;
+
+namespace DevHome.SetupFlow.ElevatedComponent;
+
+/// <summary>
+/// Class for executing tasks that run in the elevated background process.
+/// </summary>
+/// <remarks>
+/// This interface is to be extended for each kind of elevated operation we need to perform.
+/// Each method will execute an operation recognized by the elevated process.
+///
+/// The types here need to be projected using CsWinRT, so there are restrictions on them.
+/// * The types must be `public sealed class` or `public interface`.
+/// * All public members must use only types that can be projected with CsWinRT.
+/// If the projection fails, it will only produce a warning, but the build is likely to
+/// fail further down the line when attempting to use the types that should have been created.
+/// </remarks>
+public interface IElevatedComponentOperation
+{
+    /// <summary>
+    /// Writes a string to standard output.
+    /// </summary>
+    /// <remarks>
+    /// This is intended for tests only.
+    /// </remarks>
+    public void WriteToStdOut(string value);
+
+    /// <summary>
+    /// Install a package
+    /// </summary>
+    /// <param name="packageId">Package id</param>
+    /// <param name="catalogName">Package catalog name</param>
+    /// <returns>Install package operation result</returns>
+    public IAsyncOperation<ElevatedInstallTaskResult> InstallPackageAsync(string packageId, string catalogName);
+
+    /// <summary>
+    /// Create a dev drive
+    /// </summary>
+    /// <returns>Create dev drive operation hresult code</returns>
+    public int CreateDevDrive();
+
+    /// <summary>
+    /// Apply DSC configuration
```

Comment:
Nit: Mention that the dev drive created / configuration applied is the one we got in the command line args